### PR TITLE
CI: migrate pipeline into GH Actions

### PR DESCRIPTION
**Pull request by brushknight**
- [x] Review which nrjmx we inject into msi
- [x] add version print logic into codebase 
- [x] add docker login for pull rate limitations 
- [x] Add snyk step as `needs` for prerelease once it will be ok

### Update signing process
- [x] GPG private key 
- [x] GPG private key passphrase
- [x] PFX certificate
- [x] PFX certificate passphrase
- [x] update `PFX_CERTIFICATE_DESCRIPTION`

### pipeline env vars via secrets
- [x] SLACK_TOKEN
- [x] SLACK_CHANNEL

### CODE REVIEW

File: build/package/windows/nri-amd64-installer/Product.wxs [hunk 1]
```diff
@@ -0,0 +1,99 @@
+﻿<?xml version="1.0" encoding="UTF-8"?>
+<Wix xmlns="http://schemas.microsoft.com/wix/2006/wi">
+    <?define Version = "$(var.IntegrationVersion)" ?>
+    <?define ProductId = "*" ?>
+    <?define UpgradeCode = "{f2334d3a-10f1-4a4c-9185-08d197f95b2b}" ?>
```
Comment (gsanchezgavier) [inline build/package/windows/nri-amd64-installer/Product.wxs:5]: The upradedCode should be unique , i checked that the nri-redis already have this one. This should be updated before merge

**Review by gsanchezgavier — CHANGES_REQUESTED** [review]
the upgrade codes needs to be changed as it is in the comment

File: build/.goreleaser.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    bindir: "/var/db/newrelic-infra/newrelic-integrations/bin"
+    files:
+      "cassandra-config.yml.sample": "/etc/newrelic-infra/integrations.d/cassandra-config.yml.sample"
+      "CHANGELOG.md": "/usr/share/doc/nri-cassandra/CHANGELOG.md"
+      "README.md":  "/usr/share/doc/nri-cassandra/README.md"
+      "LICENSE": "/usr/share/doc/nri-cassandra/LICENSE"
+    config_files:
+      "cassandra-definition.yml": "/var/db/newrelic-infra/newrelic-integrations/cassandra-definition.yml"
+    dependencies:
+      - "newrelic-infra"
```
Comment (gsanchezgavier) [inline build/.goreleaser.yml:56]: also needs the nrjmx packages

**Review by arvdias — APPROVED** [review]

File: build/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,17 @@
+FROM golang:1.15.2-buster
+
+ARG GH_VERSION='1.4.0'
```
Comment (roobre) [inline build/Dockerfile:3]: One can never be up-to-date enough
```suggestion
ARG GH_VERSION='1.5.0'
```

**Comment by roobre** [comment]
Would it be worth to add the package testing step to this PR, rather than opening a new one today/tomorrow? Repo name should be changed but luckily we'll get the now one by today EOB.

https://github.com/newrelic/nri-redis/pull/103/files

File: .github/workflows/prerelease.yml [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        run: make ci/prerelease
+      - name: Notify failure via Slack
+        if: ${{ failure() }}
+        uses: archive/github-actions-slack@master
+        with:
+          slack-bot-user-oauth-access-token: ${{ secrets.COREINT_SLACK_TOKEN }}
+          slack-channel: ${{ secrets.COREINT_SLACK_CHANNEL }}
+          slack-text: "❌ `${{ env.REPO_FULL_NAME }}`: prerelease pipeline failed."
+      - name: Test package installability
+        uses: paologallinaharbur/test-packages-action/linux@v1
```
Comment (gsanchezgavier) [inline .github/workflows/prerelease.yml:135]: I would not add this until is in a nr repo

**Review by arvdias — APPROVED** [review]

File: .github/workflows/prerelease.yml (hunk 4 above)
Comment (arvdias) [reply .github/workflows/prerelease.yml]: done

**Review by roobre — APPROVED** [review]

File: cassandra-win-config.yml.sample [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+          # https://github.com/newrelic/infra-integrations-sdk/blob/master/docs/entity-definition.md
+          remote_monitoring: true
+          NR_JMX_TOOL: "C:\\PROGRA~1\\NEWREL~1\\nrjmx\\nrjmx.bat"
+      labels:
+          env: production
+          role: cassandra
+    - name: cassandra-inventory
+      command: inventory
+      arguments:
+          config_path: c:\apache-cassandra-3.11.9\conf\cassandra.yaml
```
Comment (gsanchezgavier) [inline cassandra-win-config.yml.sample:27]: `hostname` needs to be added for the entity name and inventory workaround.
```suggestion
          hostname: localhost
          config_path: c:\apache-cassandra-3.11.9\conf\cassandra.yaml
```

**Review by gsanchezgavier — APPROVED** [review]
looks good, just small thing in the win config for the entity naming

**Comment by roobre** [comment]
We might have to keep this open for a while until we solve the current issues with nrjmx

**Review by roobre — CHANGES_REQUESTED** [review]
We might have to keep this open until we solve the current issues with nrjmx

File: cassandra-win-config.yml.sample (hunk 5 above)
Comment (arvdias) [reply cassandra-win-config.yml.sample:27]: did not know that, thanks

**Review by arvdias — APPROVED** [review]

**Review by roobre — APPROVED** [review]
all green

**Review by arvdias — APPROVED** [review]

**Pull request merged**